extensions/kafka-connect/src/test/java/com/hazelcast/jet/kafka/connect/impl/DummySourceConnector.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.kafka.connect.impl;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectorContext;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.connect.source.SourceTaskContext;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public final class DummySourceConnector extends SourceConnector {
+
+    static final String ITEMS_SIZE = "dummy.source.connector.items.size";
+    static DummySourceConnector INSTANCE;
+    private boolean started;
+    private Map<String, String> properties;
+    private boolean initialized;
+
+    public DummySourceConnector() {
+        INSTANCE = this;
```

Comment:
why not standard singleton pattern? Escaping `this` is rarely a good idea

Reply:
The connector is instantiated here https://github.com/hazelcast/hazelcast/pull/23312/files#diff-bdf0edcf20c30bfe0e8b674136c8f53b1ee4a99f8bd43030c1b12393ed486722R55. I need INSTANCE just for test verification purposes. If you have something better in mind please provide some sample

Reply:
Right, in such usage we don't have any other options (I think).